### DISCUSSION TITLE
Fast addition of a body in Kane's method

### DISCUSSION THREAD

**Discussion by Simone2021space** [sympy/sympy#26600] [Q&A]
Hi team,
I am applying Kane's method to compute EoMs for a multibody system.
I have a list of bodies BL= [A,B,C] and forces FL=[Fa, Fb, Fc].

So I do all the necessary steps (here's just a rough outline) :
....
KM = KanesMethod(....)
(fr, frstar) = KM.kanes_equations(BL, FL)
 
Now I want to add another body D with a force Fd, so to have BL = [A,B,C,D], FL = [Fa, Fb, Fc, Fd].
Is there a way to quickly build EoMs exploiting the old model, without need to redo all the computations from scratch?

I know there are several passages to modify, like rebuild the kinematic differential equations, etc., but I am wondering if there is any way to speed up the entire process, exploiting the model I currently have.

**Comment by moorepants** [comment]
If you use the joints framework and the new System class (only available in development at the moment), you construct systems by assembling bodies. Then all the EoM setup is done behind the scenes. If you use KanesMethod directly, then you have to assemble models by deriving the correct equations yourself and that generally means a custom implementation for each model.
